kernel/api/src/main/java/org/sakaiproject/site/api/GroupLock.java

```diff
@@ -0,0 +1,48 @@
+package org.sakaiproject.site.api;
+
+import java.io.Serializable;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "SAKAI_GROUP_LOCK")
+@NamedQueries({
+	@NamedQuery(name = "findGroupWithLock", query = "from GroupLock where groupId = :groupId and lockMode = :groupLockProperty"),
+	@NamedQuery(name = "findItemsLockingGroup", query = "select itemId from GroupLock where groupId = :groupId"),
+	@NamedQuery(name = "findGroupsLockedByItem", query = "select itemId from GroupLock where itemId = :itemId")
+})
+
+@Data
+@NoArgsConstructor
+@IdClass(GroupLock.class)
```

Comment:
Is this a strange use of @IdClass usually its another class not the same

Reply:
If all the values of the class match with the id, it can be used this way.